sway-core/src/type_system/engine.rs

```diff
@@ -172,22 +172,17 @@ impl TypeEngine {
                     });
                     return err(warnings, errors);
                 }
-                for type_argument in type_arguments.iter_mut() {
-                    type_argument.type_id = check!(
-                        self.resolve(
-                            decl_engine,
-                            type_argument.type_id,
-                            &type_argument.span,
-                            enforce_type_arguments,
-                            None,
-                            namespace,
-                            mod_path
-                        ),
-                        self.insert(decl_engine, TypeInfo::ErrorRecovery),
-                        warnings,
-                        errors
-                    );
-                }
+
+                self.resolve_type_arguments(
```

Comment:
Note my other comment about restructuring the `resolve_type_arguments` method. --- if the changes from that comment are implemented, then at this call site you can use the `check!` macro again like so:

```rust
check!(
  self.resolve_type_arguments(......),
  return err(warnings, errors),
  warnings,
  errors,
);
```

This ensures that we only return will `err` if the whole of `resolve_type_arguments` returns an error.

However---the previous source code didn't account for this, so this request is just a code style one really 😃 I'm happy to discuss.